icu4c/source/i18n/coptccal.cpp

```diff
@@ -62,22 +63,19 @@ CopticCalendar::handleGetExtendedYear(UErrorCode& status)
     if (U_FAILURE(status)) {
         return 0;
     }
-    int32_t eyear;
     if (newerField(UCAL_EXTENDED_YEAR, UCAL_YEAR) == UCAL_EXTENDED_YEAR) {
-        eyear = internalGet(UCAL_EXTENDED_YEAR, 1); // Default to year 1
-    } else {
-        // The year defaults to the epoch start, the era to CE
-        int32_t era = internalGet(UCAL_ERA, CE);
-        if (era == BCE) {
-            eyear = 1 - internalGet(UCAL_YEAR, 1); // Convert to extended year
-        } else if (era == CE){
-            eyear = internalGet(UCAL_YEAR, 1); // Default to year 1
-        } else {
-            status = U_ILLEGAL_ARGUMENT_ERROR;
-            return 0;
-        }
+        return internalGet(UCAL_EXTENDED_YEAR, 1); // Default to year 1
+    }
+    // The year defaults to the epoch start, the era to CE
+    int32_t era = internalGet(UCAL_ERA, CE);
+    if (era == BCE) {
+        return 1 - internalGet(UCAL_YEAR, 1); // Convert to extended year
     }
-    return eyear;
+    if (era == CE){
+        return internalGet(UCAL_YEAR, 1); // Default to year 1
+    }
+    status = U_ILLEGAL_ARGUMENT_ERROR;
+    return 0;
```

Comment:
I just have to editorialize for a minute here (I'm not really asking you to change anything).

For years, I worked on a team that was very dogmatic about having one return statement per function and regularly got slapped around for having more than one return, and so seeing a bunch of code in here where you're doing the exact opposite and preferring mid-function returns over else clauses just makes me twitch uncontrollably.

My own preference is somewhere in the middle-- I like if statements to be balanced most of the time.  I think it improves readability and helps protect against early returns accidentally bypassing needed cleanup code.  But I also think always requiring a single return can really mess up readability in a lot of functions for more complex logic.  

To my knowledge ICU doesn't have a set of style guidelines that mandate anything in particular.  You're the principal maintainer of these files now, so I guess you should do it however you find it the most comfortable.  But I flinch a little at changes like this that are all about coding style.

Reply:
> having one return statement per function

That will work great if we ALSO ensure if all functions are < 20 lines (including comment line), one TAB = 2 chars, and break longer function to several smaller functions. With that two constraits together, I will also prefer one return statement per function. The problem in ICU code right now is we have TAB=4 chars and we currently have a lot of 
```
if (U_FAILURE(status)) {
   return;
}
// 10 lines of code
if (U_FAILURE(status)) {
   return;
}
// another 3 pages of code
return
```
if we maintain the "one return statement per function" style, then it will become

```
if (U_SUCCESS(status)) {
  // 10 lines of code
  if (U_SUCECESS(status)) {
      // another 3 pages of code
  }
}
return
```
and the if block is keep indenting and moving right and greater than 80 col per line. (The fact we use TAB=4 chars make the block moving left x2 than TAB=2)

Reply:
Yeah, I understand.  I just keep flashing back to the beatings I got on a previous team when I did that kind of thing.  :-)